From 572faa5c2b472c2842769667ecfc937d2f9f23b7 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:01 -0400
Subject: [fs] btrfs: move btrfs_{set, clear}_and_info() to ctree.h

Message-id: <1407179109-13714-295-git-send-email-sandeen@redhat.com>
Patchwork-id: 87257
O-Subject: [RHEL7.1 PATCH 294/422] Btrfs: move btrfs_{set, clear}_and_info() to ctree.h
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit 9d89ce658718d9d21465666127a15ca2c82c4ea7
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Wed Apr 23 19:33:33 2014 +0800

    Btrfs: move btrfs_{set,clear}_and_info() to ctree.h

    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ctree.h b/fs/btrfs/ctree.h
index ad1a594..e7c9e1c 100644
--- a/fs/btrfs/ctree.h
+++ b/fs/btrfs/ctree.h
@@ -2058,6 +2058,20 @@ struct btrfs_ioctl_defrag_range_args {
 #define btrfs_raw_test_opt(o, opt) ((o) & BTRFS_MOUNT_##opt)
 #define btrfs_test_opt(root, opt) ((root)->fs_info->mount_opt & \
       BTRFS_MOUNT_##opt)
+#define btrfs_set_and_info(root, opt, fmt, args...)   \
+{         \
+ if (!btrfs_test_opt(root, opt))     \
+  btrfs_info(root->fs_info, fmt, ##args);   \
+ btrfs_set_opt(root->fs_info->mount_opt, opt);   \
+}
+
+#define btrfs_clear_and_info(root, opt, fmt, args...)   \
+{         \
+ if (btrfs_test_opt(root, opt))     \
+  btrfs_info(root->fs_info, fmt, ##args);   \
+ btrfs_clear_opt(root->fs_info->mount_opt, opt);   \
+}
+
 /*
  * Inode flags
  */
diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index 53bc373..363404b 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -385,20 +385,6 @@ static match_table_t tokens = {
  {Opt_err, NULL},
 };
 
-#define btrfs_set_and_info(root, opt, fmt, args...)   \
-{         \
- if (!btrfs_test_opt(root, opt))     \
-  btrfs_info(root->fs_info, fmt, ##args);   \
- btrfs_set_opt(root->fs_info->mount_opt, opt);   \
-}
-
-#define btrfs_clear_and_info(root, opt, fmt, args...)   \
-{         \
- if (btrfs_test_opt(root, opt))     \
-  btrfs_info(root->fs_info, fmt, ##args);   \
- btrfs_clear_opt(root->fs_info->mount_opt, opt);   \
-}
-
 /*
  * Regular mount options parser.  Everything that is needed only when
  * reading in a new superblock is parsed here.
-- 
1.7.1